src/firebase-config-common.ts

```diff
@@ -0,0 +1,19 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type CommonDocumentData = { [field: string]: any };
+/** An interface for `FirestoreDataConverter` common to both frontend firebase and admin firebase. */
+export interface CommonFirestoreDataConverter<T> {
+  toFirestore(modelObject: T): CommonDocumentData;
+  fromFirestore(snapshot: { data(): CommonDocumentData }): T;
+}
+
+export const getTypedFirestoreDataConverter = <T>(): CommonFirestoreDataConverter<T> => ({
+  fromFirestore(snapshot) {
+    return snapshot.data() as T;
+  },
+  toFirestore(userData) {
+    return userData;
+  },
```

Comment:
If you look at the old code, all the type converters look exactly the same. In fact, they are completely the same when the TS code is type erased to JS. Therefore, we can factor them out into a helper constructor, and use the passed-in generic type parameter to figure the type out!